frontend/sentiment-dashboard/src/app/dashboard/dashboard.component.ts

```diff
@@ -72,6 +80,22 @@ export class DashboardComponent implements OnInit, OnDestroy {
             });
   }
 
+  public updateStatsSubscription2(event: StatsFilterValues) {
```

Comment:
Have you checked what happens when we're not comparing? Just wanted to make sure there weren't an errors when statsCompare isn't visible/undefined.